targettests/execution/qubit_management_if-1.cpp

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include <cudaq.h>
+
+// TODO: filecheck with statistics
+
+// RUN: nvq++ --target opt-test --target-option dep-analysis,qpp %s -o %t && %t
+
+// Simple test, shouldn't affect anything
+struct run_test {
+  __qpu__ auto operator()() {
```

Comment:
```suggestion
  __qpu__ auto operator()(bool condition) {
```
just wondering if removing the constant condition breaks anything

Reply:
(Same as above.) Will create a separate issue for this one (or add to a new grouped issue).

Reply:
https://github.com/NVIDIA/cuda-quantum/issues/2181 written. Resolving this conversation.